### DISCUSSION TITLE
M5 Paper - hackable e-ink device

### DISCUSSION THREAD

**Discussion by gingerbeardman** [koreader/koreader#10809] [Ideas]
The M5 Paper is a hackable touch-screen e-ink display with Arduino-compatible CPU, wifi and a nice little rocker switch that could turn pages. It's smartphone-sized.

There are epub readers available for it, [example](https://github.com/turgu1/EPub-InkPlate), but none are as good as KOReader.

If there is any interest from a KOReader developer, I would be happy to loan mine out.

https://docs.m5stack.com/en/core/m5paper

![M5 Paper](https://static-cdn.m5stack.com/resource/docs/products/core/m5paper/m5paper_01.webp)

**Comment by NiLuJe** [comment]
You probably won't be able to do anything w/ KOReader on such a memory constrained environment (you might have been able to get away with just the very slow CPU, though).

**Comment by pazos** [comment]
It isn't possible to run KOReader at all.

LuaJIT wasn't ported to Xtensa.
Even with a port in place the device isn't powerful enough to do just in time compilation and will waste most of the available ram to hold JIT compiled lua modules.
Even then there's no dynamic library loading.

The bare minimum requirements for hardware to run KOReader on are:

- A CPU with a MMU supported by both the Linux kernel and LuaJIT
- 128MB of RAM (it could be way less with virtual memory & disk swapping, but too slow to be considered useful)
- About 300MB of disk.

**Reply by gingerbeardman** [reply]
Thanks for the info!

**Reply by 10a5er** [reply]
🥲
